Modules/Core/Common/include/itkPointSet.hxx

```diff
@@ -81,6 +81,50 @@ PointSet<TPixelType, VDimension, TMeshTraits>::SetPoints(PointsVectorContainer *
   this->Modified();
 }
 
+
+template <typename TPixelType, unsigned int VDimension, typename TMeshTraits>
+void
+PointSet<TPixelType, VDimension, TMeshTraits>::SetPointsByCoordinates(const std::vector<CoordRepType> & coordinates)
+{
+  itkDebugMacro("Setting the points to the specified coordinates");
+
+  const size_t numberOfCoordinates = coordinates.size();
+
+  if (numberOfCoordinates % PointDimension != 0)
+  {
+    itkExceptionMacro("Number of specified coordinates incompatible with the point dimension");
+  }
+
+  const size_t numberOfPoints = numberOfCoordinates / PointDimension;
+
+  if (m_PointsContainer == nullptr)
+  {
+    m_PointsContainer = PointsContainer::New();
+  }
+
+  using STLContainerType = typename PointsContainer::STLContainerType;
+
+  STLContainerType & points = m_PointsContainer->CastToSTLContainer();
+  points.clear();
+
+  if constexpr (std::is_same_v<STLContainerType, std::vector<PointType>>)
+  {
+    points.resize(numberOfPoints);
```

Comment:
Why is this inside an `if`? Isn't it always needed?

Reply:
Good question... as far as I observed, the STLContainerType of the PointsContainer is either an std::vector or an std::map. If it is an std::map, the expression `points[pointIdentifier]` just works well without `points.resize(numberOfPoints)`. And actually, std::map does not have a resize member function.

Should there be an extra line of comment, explaining why?

Reply:
```suggestion
    points.resize(numberOfPoints);  // std::map does not have a resize function
```

Reply:
Yes indeed, something like that, thanks. I'll do a little amend + force-push.

Reply:
Ideally the code should be more like:

```
if constexpr ( `points.resize(numberOfPoints)` is a valid expression )
{
  points.resize(numberOfPoints);
}
```

But I found it easier to just write `if constexpr (std::is_same_v<STLContainerType, std::vector<PointType>>)`  😃

Reply:
Please check my [force-push](https://github.com/InsightSoftwareConsortium/ITK/compare/37e1b83e065f07264e55450a55d02ba5db376738..9cb3f7c7ffdb76c1efcd9077e35aaf1324a70935)  😃

Reply:
By the way, I could still add an `else` clause, having `static_assert(std::is_same_v<STLContainerType, std::map>)`. If anyone would like that!

Reply:
@dzenanz Here it is, please have a look: [force-push](https://github.com/InsightSoftwareConsortium/ITK/compare/9cb3f7c7ffdb76c1efcd9077e35aaf1324a70935..60d183c0d6e12034f61dcfaa52e95e90bca8290e)